src/sentry/grouping/ingest/hashing.py

```diff
@@ -174,43 +172,8 @@ def run_primary_grouping(
     Get the primary grouping config and primary hashes for the event.
     """
     with metrics.timer("event_manager.load_grouping_config"):
-        if is_reprocessed_event(job["data"]):
-            # The customer might have changed grouping enhancements since
-            # the event was ingested -> make sure we get the fresh one for reprocessing.
-            grouping_config = get_grouping_config_dict_for_project(project)
-            # Write back grouping config because it might have changed since the
-            # event was ingested.
-            # NOTE: We could do this unconditionally (regardless of `is_processed`).
-            job["data"]["grouping_config"] = grouping_config
-        else:
-            grouping_config = get_grouping_config_dict_for_event_data(
-                job["event"].data.data, project
-            )
-
-            # TODO: For new (non-reprocessed) events, we read the grouping config off the event
-            # rather than from the project. But that grouping config is put there by Relay after
-            # looking it up on the project. Are these ever not the same? If we don't ever see this
-            # log, after some period of time we could probably just decide to always follow the
-            # behavior from the reprocessing branch above. If we do that, we should decide if we
-            # also want to stop adding the config in Relay.
-            # See https://github.com/getsentry/sentry/pull/65116.
-            config_from_relay = grouping_config["id"]
-            config_from_project = project.get_option("sentry:grouping_config")
-
-            if config_from_relay != config_from_project:
-                # The relay value might not match the value stored on the project if the project was
-                # recently updated and relay's still using its cached value. Based on logs, this delay
-                # seems to be about 3 seconds, but let's be generous and give it a minute to account for
-                # clock skew, network latency, etc.
-                if not _config_update_happened_recently(project, 30):
-                    logger.info(
-                        "Event grouping config different from project grouping config",
-                        extra={
-                            "project": project.id,
-                            "relay_config": config_from_relay,
-                            "project_config": config_from_project,
-                        },
-                    )
+        grouping_config = get_grouping_config_dict_for_project(project)
+        job["data"]["grouping_config"] = grouping_config
```

Comment:
Is there value in keeping this comment?
```
            # Write back grouping config because it might have changed since the
            # event was ingested.
```

Your call.

Reply:
I debated that. In the context of "Normally we don't write config data to the event because Relay has already done it for us, but in this specific case we want to, and here's why," it definitely made sense to have it there. But in this new world it's just "When we do grouping, we record on the event the config being used," and so I felt like it was actually over-complicating things to call out the reprocessing case specifically.

Reply:
👍🏻